### PR TITLE
refactor: Config migration sonarlint issues

### PR DESCRIPTION
**Pull request by MichaelsJP**
This PR includes several sonar issues that popped up while working on the config migration. Often times it is not new code, but code that got touched and re-triggered.

This PR avoids bigger code refactors. Some smaller ones were made, where I felt confident to do so.

A bit bigger code refactoring was necessary in [ExtendedStorageProperties.java](https://github.com/GIScience/openrouteservice/pull/1910/files#diff-e8d84c2c1aef6902a546d3d8ed96baca8c85c02663c322c32a384d22935b3d60).
I was part of creating that class, so I feel confident in introducing those changes. The class was already at a decent test coverage and now is at 100%.

### Pull Request Checklist

- [ ] 1. I have [**rebased**][rebase] the latest version of the main branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes # .

### Information about the changes
- Key functionality added:
- Reason for change:

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/run-instance/configuration/
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/main/CONTRIBUTE.md#pull-request-guidelines

### CODE REVIEW

File: ors-engine/src/main/java/org/heigit/ors/config/profile/ExtendedStorageProperties.java [hunk 1]
```diff
@@ -159,111 +160,138 @@ public void initialize(ExtendedStorageName storageName) {
...
+    private void initializeHeavyVehicle(ArrayList<String> nonNullableProperties) {
+        if (restrictions == null) {
+            restrictions = true;
         }
-        setAllPropertiesToNull(nonNullableProperties);
+        nonNullableProperties.add("restrictions");
+    }
+
+    private void initializeHillIndex(ArrayList<String> nonNullableProperties) {
+        nonNullableProperties.add("maximum_slope");
```
Comment (jhaeu) [inline ors-engine/src/main/java/org/heigit/ors/config/profile/ExtendedStorageProperties.java:200]: Is omitting 
```java
if (maximumSlope == null) {
   this.maximumSlope = null;
}
```
intended?

File: ors-engine/src/main/java/org/heigit/ors/config/utils/NonEmptyMapFilter.java [hunk 2]
```diff
@@ -18,4 +19,9 @@ public boolean equals(Object other) {
         }
         return false;
     }
+
+    @Override
+    public int hashCode() {
```
Comment (jhaeu) [inline ors-engine/src/main/java/org/heigit/ors/config/utils/NonEmptyMapFilter.java:18]: Would lombok's @EqualsAndHashCode be an option, too?

**Review by jhaeu — COMMENTED** [review]
Well done - especially thanks to the cleanup of ExecutionProperties!!!
Please have a look at the two comments.

File: ors-engine/src/main/java/org/heigit/ors/config/profile/ExtendedStorageProperties.java (hunk 1 above)
Comment (MichaelsJP) [reply ors-engine/src/main/java/org/heigit/ors/config/profile/ExtendedStorageProperties.java:200]: Yes. The call was redundant. maximumSlope isn't a scoped property but always the global one. So `maximumSlope` and `this.maximumSlope` reference the same property. :smile:

File: ors-engine/src/main/java/org/heigit/ors/config/utils/NonEmptyMapFilter.java (hunk 2 above)
Comment (MichaelsJP) [reply ors-engine/src/main/java/org/heigit/ors/config/utils/NonEmptyMapFilter.java:18]: Unfortunately not. Even with Lombok, you have to take `equals` and `HashCode` from the same source. In this case, we manually override Equals and, therefore also for Lombok need to manually add `hashCode`.

But your comment made me revisit the Equals function. It is using reflection :unamused: . Will quickly refactor it.

File: ors-engine/src/main/java/org/heigit/ors/routing/RoutingProfile.java [hunk 3]
```diff
@@ -526,11 +523,9 @@ public void setSpeedups(GHRequest req, boolean useCH, boolean useCore, boolean u
             req.setAlgorithm(Parameters.Algorithms.DIJKSTRA_BI);
             req.setProfile(profileNameCH);
         }
-        if (useCore) {
-            // fallback to a core profile without turn costs if one is available
```
Comment (koebi) [inline ors-engine/src/main/java/org/heigit/ors/routing/RoutingProfile.java:635]: Why is the comment omitted?

**Review by koebi — COMMENTED** [review]
Apart from the Reflection refactoring, which I don't know about, looks good to me :)

File: ors-engine/src/main/java/org/heigit/ors/routing/RoutingProfile.java (hunk 3 above)
Comment (MichaelsJP) [reply ors-engine/src/main/java/org/heigit/ors/routing/RoutingProfile.java:635]: You mean why I didn't delete it?

Comment (MichaelsJP) [reply ors-engine/src/main/java/org/heigit/ors/routing/RoutingProfile.java:635]: Hm. I guess  the IntelliJ magic just ate it. Will add it back.

Comment (MichaelsJP) [reply ors-engine/src/main/java/org/heigit/ors/routing/RoutingProfile.java:635]: Fixed!

**Review by jhaeu — APPROVED** [review]
LGTM

**Pull request merged**